File: 0125-valid-palindrome/0125-valid-palindrome.c

bool isPalindrome(char* s) {
    int len = strlen(s);
    int j=0;
    for(int i=0;i<len;i++)
    {
        
        if(isalpha(s[i])||isdigit(s[i]))
        {
            s[j++]=tolower(s[i]);
        }

    }
    s[j]='\0';
    int r=0,l=j-1;
    while(r<l)
    {
        if(s[r++]!=s[l--]){
            return false;
        }
    }
    return true;
}

File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        
        s = s.toLowerCase();
        StringBuilder res = new StringBuilder();
        for(char i:s.toCharArray()){
            if(Character.isLetter(i) || Character.isDigit(i)){
                res.append(i);
            }
        }
        StringBuilder temp = new StringBuilder(res);
        temp=temp.reverse();
        for(int i=0;i<res.length();i++){
            if(res.charAt(i) != temp.charAt(i)) return false;
        }
        return true;
    }
}